app/waf/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# build with:
+#   docker build . -t f5devcentral/f5-wafaas-openstack:latest
+
+# upload image with (authorized user only)
+#   docker push f5devcentral/f5-wafaas-openstack:latest
+
+FROM node:11.6.0
+
+LABEL maintainer="F5 Networks Inc. Corp."
+EXPOSE 3000/tcp
+
+WORKDIR /root/app_dir
+ADD dist dist
+RUN rm -r dist/test
+ADD public public
+COPY package.json index.js ./
+RUN npm install -g nc
+RUN npm install
+
+ENTRYPOINT \
+    timeout=15; \
+    while [ $timeout -gt 0 ]; do \
```

Comment:
I think every app image should be independent and as simple as much. It may not assume any dependencies. Those dependencies should be configurable, and deployment tool will take care those dependencies.

Reply:
After a day of investigation, DataSource's EventEmitter would post error immediately after any action to db datasource, even get: `app.get<DataSource>('datasources.db')`. So still need more knowledge about the datasource usage, and try to optimize it in future. 

Error output: 

```
Connection fails: Error: connect ECONNREFUSED 127.0.0.1:5432
It will be retried for the next request.
events.js:173
      throw er; // Unhandled 'error' event
      ^

Error: connect ECONNREFUSED 127.0.0.1:5432
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1083:14)
Emitted 'error' event at:
```